test/Sentry.AspNetCore.Tests/allsettings.json

```diff
@@ -8,8 +8,14 @@
     "MinimumEventLevel": "Critical",
     "InitializeSdk": "false",
     "MaxBreadcrumbs": "999",
-    "SampleRate": "1",
+    "SampleRate": 1.0,
     "Release": "7f5d9a1",
-    "Environment": "Staging"
+    "Environment": "Staging",
+    "TracesSampleRate": 1.0,
+    "TracePropagationTargets": [
+      "foo",
+      "bar",
+      "regex:^abc.*ghi$"
```

Comment:
I hope this `regex:` prefix makes sense?  I didn't see any other simple way to add these to a json config file and still be able to distinguish between regex and substring.

Reply:
Can you rather do an internal try catch to figure out if its a valid Regex or not?
On Dart, we can do catch the `FormatException` type.

Reply:
I actually didn't handle that myself https://github.com/getsentry/sentry-dart/pull/1043/files

Reply:
Per our discussion, I will update the behavior to:

- Remove need for prefix
- Always try substring matching first
- If fails substring matching, try regex matching